tools/ensure_codegen_up_to_date.py

```diff
@@ -0,0 +1,64 @@
+"""
+ensure_codegen_up_to_date
+-------------------------
+Ensure changes to code generation are committed to repository
+"""
+
+import os
+import sys
+import subprocess
+
+def configure_git_user_email():
+    """
+    Configures git user email with dummy email id
+    """
+    if os.system('git config user.email "dummy@ni.com"') != 0:
+        print("Unable to configure \"user email\" using git", file=sys.stderr)
+        sys.exit(-1)
+
+def configure_git_user_name():
+    """
+    Configures git user email with dummy name
+    """
+    if os.system('git config user.name "dummy"') != 0:
+        print("Unable to configure \"user name\" using git", file=sys.stderr)
+        sys.exit(-1)
+
+def clean_codegen_files():
+    """
+    Before code generation clean the existing codegen files
+    """
+    if os.system("tox -e clean") != 0:
+        print("Unable to clean the repo using \"tox -e clean\"", file=sys.stderr)
+        sys.exit(-1)
+
+def create_codegen_files():
+    """
+    create codegen files
+    """
+    if os.system("tox -e codegen") != 0:
+        print("Unable to generate code using \"tox -e codegen\"", file=sys.stderr)
+        sys.exit(-1)
+
+def check_file_status():
```

Comment:
The name of this function does not describe what the function does. That's bad. 
I myself aren't sure what it does. It appears to "return the number of modified uncommitted files".
If that's the case, then you should rename it. Something like `get_number_of_modified_files` or something.

But I think it's better to instead change the function to `check_no_dirty_files` and have it return the error message and error inside the body of the function - because that's what you do in all the other functions.

Reply:
Do you mean that the check_no_dirty_files function would have this section in its body instead of in main?
`print("Run codegen and include the generated files in the PR", file=sys.stderr)
        sys.exit(-1)`

Reply:
Yes, so that `__main__` is straightforward and readable and all the helper methods are where the `sys.exit` occurs.

Reply:
Updated